core/src/subgraph/instance_manager.rs

```diff
@@ -218,18 +218,28 @@ impl<S: SubgraphStore> SubgraphInstanceManager<S> {
 
             info!(logger, "Successfully resolved subgraph files using IPFS");
 
-            // Add dynamic data sources to the subgraph
-            manifest.data_sources.extend(data_sources);
-
             info!(
                 logger,
                 "Data source count at start: {}",
                 manifest.data_sources.len()
             );
 
-            (manifest, manifest_idx_and_name)
+            let static_data_sources = manifest.data_sources.clone();
+
+            // Add dynamic data sources to the subgraph
+            manifest.data_sources.extend(data_sources);
+
+            (manifest, manifest_idx_and_name, static_data_sources)
         };
 
+        let dds_count = manifest
+            .data_sources
+            .len()
+            .abs_diff(static_data_sources.len());
```

Comment:
I wouldn't bother doing the diff and would just use the total len but this is also fine.